cpp/harvest/marc_issn_lookup.cc

```diff
@@ -0,0 +1,190 @@
+/**
+ * \brief Utility for updating issn information
+ * \author Steven Lolong (steven.lolong@uni-tuebingen.de)
+ *
+ * \copyright 2023 Tübingen University Library.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <fstream>
+#include <iostream>
+#include "CORE.h"
+#include "FileUtil.h"
+#include "MARC.h"
+#include "util.h"
+
+namespace {
+[[noreturn]] void Usage() {
+    ::Usage(
+        "marc_input_articles marc_input_journals marc_output_articles"
+        "\n"
+        "- marc_input_articles is a file containing all article information taken from CORE.\n"
+        "- marc_input_journals is a file containing journal information. Please use issn_lookup.py to generate this file.\n"
+        "- marc_output_articles is an output file generated by this tool.\n");
+
+    std::exit(EXIT_FAILURE);
+}
+struct SubFieldInfo {
+    std::string i_;
+    std::string t_;
+    std::string w_;
+    std::string x_;
+    int online_version_counter_;
+    int printed_version_counter_;
+    bool is_online_;
+    bool is_valid_;
+
+
+    void Builder(MARC::Record &record) {
+        for (auto &field : record) {
+            if (field.getTag() == "001")
+                w_ = "(DE-627)" + field.getContents();
+
+            if (field.getTag() == "245") {
+                MARC::Subfields subfields(field.getSubfields());
+                std::string subfield_a(subfields.getFirstSubfieldWithCode('a'));
+                std::string subfield_b(subfields.getFirstSubfieldWithCode('b'));
+                if (subfield_a != "" && subfield_b != "")
+                    t_ = subfield_a + " " + subfield_b;
+                else if (subfield_a != "" && subfield_b == "")
+                    t_ = subfield_a;
+                else if (subfield_a == "" && subfield_b != "")
+                    t_ = subfield_b;
+                else
+                    t_ = "";
+            }
+
+            if (field.getTag() == "300")
+                ((field.getFirstSubfieldWithCode('a') == "Online-Ressource") ? is_online_ = true : is_online_ = false);
+
+
+            if (field.getTag() == "773")
+                x_ = field.getFirstSubfieldWithCode('x');
+        }
+
+        i_ = "In:";
```

Comment:
Do we really need to store `i_` in the cache if it's always the same hardcoded value `In:` for all thousands of entries? (remove memory usage and move hardcoded value to the `UpdateSubfield` function)